capsules/src/lsm303dlhc.rs

```diff
@@ -0,0 +1,779 @@
+//!     - `6` - Z acceleration in m/s2 (not scaled)
+//!     - `8` - Z magnetometer in Gauss (not scaled)
+//!
+//! Usage
+//! -----
+//!
+//! ```rust
+//! let mux_i2c = components::i2c::I2CMuxComponent::new(&stm32f3xx::i2c::I2C1)
+//!     .finalize(components::i2c_mux_component_helper!());
+//!
+//! let lsm303dlhc = components::lsm303dlhc::Lsm303dlhcI2CComponent::new()
+//!    .finalize(components::lsm303dlhc_i2c_component_helper!(mux_i2c));
+//!
+//! lsm303dlhc.configure(
+//!    lsm303dlhc::Lsm303dlhcAccelDataRate::DataRate25Hz,
+//!    false,
+//!    lsm303dlhc::Lsm303dlhcScale::Scale2G,
+//!    false,
+//!    true,
+//!    lsm303dlhc::Lsm303dlhcMagnetoDataRate::DataRate3_0Hz,
+//!    lsm303dlhc::Lsm303dlhcRange::Range4_7G,
+//!);
+//! ```
+//!
+//! NideDof Example
+//!
+//! let grant_cap = create_capability!(capabilities::MemoryAllocationCapability);
+//! let grant_ninedof = board_kernel.create_grant(&grant_cap);
+//!
+//! // use as primary NineDof Sensor
+//! let ninedof = static_init!(
+//!    capsules::ninedof::NineDof<'static>,
+//!    capsules::ninedof::NineDof::new(lsm303dlhc, grant_ninedof)
+//! );
+//!
+//! hil::sensors::NineDof::set_client(lsm303dlhc, ninedof);
+//!
+//! // use as secondary NineDof Sensor
+//! let lsm303dlhc_secondary = static_init!(
+//!    capsules::ninedof::NineDofNode<'static, &'static dyn hil::sensors::NineDof>,
+//!    capsules::ninedof::NineDofNode::new(lsm303dlhc)
+//! );
+//! ninedof.add_secondary_driver(lsm303dlhc_secondary);
+//! hil::sensors::NineDof::set_client(lsm303dlhc, ninedof);
+//!
+//! ```
+//!
+//! Temperature Example
+//!
+//! ```rust
+//! let grant_cap = create_capability!(capabilities::MemoryAllocationCapability);
+//! let grant_temp = board_kernel.create_grant(&grant_cap);
+//!
+//! lsm303dlhc.configure(
+//!    lsm303dlhc::Lsm303dlhcAccelDataRate::DataRate25Hz,
+//!    false,
+//!    lsm303dlhc::Lsm303dlhcScale::Scale2G,
+//!    false,
+//!    true,
+//!    lsm303dlhc::Lsm303dlhcMagnetoDataRate::DataRate3_0Hz,
+//!    lsm303dlhc::Lsm303dlhcRange::Range4_7G,
+//!);
+//! let temp = static_init!(
+//! capsules::temperature::TemperatureSensor<'static>,
+//!     capsules::temperature::TemperatureSensor::new(lsm303dlhc, grant_temperature));
+//! kernel::hil::sensors::TemperatureDriver::set_client(lsm303dlhc, temp);
+//!
+//! ```
+//!
+//! Author: Alexandru Radovici <msg4alex@gmail.com>
+//!
+
+use core::cell::Cell;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+use kernel::common::cells::{OptionalCell, TakeCell};
+use kernel::debug;
+use kernel::hil::i2c::{self, Error};
+use kernel::hil::sensors;
+use kernel::{AppId, Callback, Driver, ReturnCode};
+
+/// Syscall driver number.
+use crate::driver;
+pub const DRIVER_NUM: usize = driver::NUM::Lsm303dlch as usize;
+
+// Buffer to use for I2C messages
+pub static mut BUFFER: [u8; 8] = [0; 8];
+
+/// Register values
+const REGISTER_AUTO_INCREMENT: u8 = 0x80;
+
+/// Accelerometer Registers
```

Comment:
I think these might also benefit from using `enum_from_primitive`, then I think you wouldn't have to have the presently unused fellows commented out, and this `///` doc comment would document all of the enum members rather than this one constant

Reply:
Done

Reply:
Done.